Error initializating orderedSeq struct in ordered_presentation/c++ example subscriber

### Description

- **GCC version**: 8.3.1 20190223
- **Operating system**: Fedora 29

### What is the current behavior?

Error initializing `data_seq` variable in ordered_presentation c++ subscriber in line [75](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx#L75):

```plaintext
/home/luis/git/rticonnextdds-examples-fork/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx: In function ‘void poll_data(orderedDataReader**, int)’:
/home/luis/git/rticonnextdds-examples-fork/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx:75:31: error: could not convert ‘{1, 0, 0, 0, 0, 29508, 0, 0, {1, 0, 1}, {1, 1}, 2147483647}’ from ‘<brace-enclosed initializer list>’ to ‘orderedSeq’
         orderedSeq data_seq = DDS_SEQUENCE_INITIALIZER;
                               ^~~~~~~~~~~~~~~~~~~~~~~~
```

Warning initializing `profile_name` variable in ordered_presentation c++ subscriber in lines [151 and 152](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx#L151-L152):

```plaintext
/home/luis/git/rticonnextdds-examples-fork/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx: In function ‘int subscriber_main(int, int)’:
/home/luis/git/rticonnextdds-examples-fork/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx:152:48: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
             "ordered_Profile_subscriber_topic" };
                                                ^
```

--- a/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx
+++ b/examples/connext_dds/ordered_presentation/c++/ordered_subscriber.cxx
@@ -72,7 +72,7 @@ void poll_data(orderedDataReader *ordered_reader[], int numreaders) {
     DDS_ReturnCode_t retcode = DDS_RETCODE_OK;
     for (int r = 0; r < numreaders; ++r) {
         DDS_SampleInfoSeq info_seq;
-        orderedSeq data_seq = DDS_SEQUENCE_INITIALIZER;
+        orderedSeq data_seq;
         retcode = ordered_reader[r]->take(data_seq, info_seq,
                 DDS_LENGTH_UNLIMITED, DDS_ANY_SAMPLE_STATE, DDS_ANY_VIEW_STATE,
                 DDS_ANY_INSTANCE_STATE);
@@ -148,8 +148,8 @@ extern "C" int subscriber_main(int domainId, int sample_count) {
     DDS_Duration_t receive_period = { 4, 0 };
     int status = 0;
     int i = 0;
-    char* profile_name[] = { "ordered_Profile_subscriber_instance",
-            "ordered_Profile_subscriber_topic" };
+    char* profile_name[] = { (char*)"ordered_Profile_subscriber_instance",
+            (char*) "ordered_Profile_subscriber_topic" };
 
     /* To customize the participant QoS, use
      the configuration file USER_QOS_PROFILES.xml */

